src/solvers/dgmulti/flux_differencing.jl

```diff
@@ -141,25 +193,60 @@ function entropy_projection!(cache, u, mesh::VertexMappedMesh, equations, dg::DG
   entropy_projected_u_values .= entropy2cons.(projected_entropy_var_values, equations)
 end
 
-function calc_volume_integral!(du, u::StructArray, volume_integral,
+function calc_volume_integral!(du, u, volume_integral,
+                               mesh::VertexMappedMesh, equations, dg::DGMultiFluxDiff{<:SBP}, cache)
+
+  rd = dg.basis
+  @unpack fluxdiff_local_threaded, sparsity_pattern = cache
+  @unpack volume_flux = volume_integral
+
+  # Todo: simplices. Dispatch on curved/non-curved mesh types, this code only works for affine meshes (accessing rxJ[1,e],...)
+  @threaded for e in eachelement(mesh, dg, cache)
+    fluxdiff_local = fluxdiff_local_threaded[Threads.threadid()]
+    fill!(fluxdiff_local, zero(eltype(fluxdiff_local)))
+    u_local = view(u, :, e)
+    for i in eachdim(mesh)
+      Qi_skew_Tr = build_lazy_physical_derivative(e, i, mesh, dg, cache)
+
+      # use `sparsity_pattern` to dispatch hadamard_sum_A_transposed. If using Tri or Tet elements,
+      # the Qi_skew_Tr matrices are dense, and sparsity_pattern = nothing. If using Quad or Hex
+      # elements with an SBP approximationType, then sparsity_pattern::AbstractSparseMatrix{Bool}.
+      hadamard_sum_A_transposed!(fluxdiff_local, Qi_skew_Tr, volume_flux, i,
+                                 u_local, sparsity_pattern,
+                                 mesh, equations, dg, cache)
+    end
+
+    for i in each_quad_node(mesh, dg, cache)
+      du[i, e] = du[i, e] + fluxdiff_local[i] / rd.wq[i]
```

Comment:
It might be worth considering to move the division by the weights `rd.wq[i]` into the definition of the matrices used to compute the flux differencing stuff. At least that's what we use for our other solvers - storing basically `2 * D - M \ B`, where `M` is the mass matrix, `D` the differentiation matrix, and `B` the boundary matrix. This avoids additional (slow) divisions.

Reply:
That's a good point. I'll make one more modification to make this clearer (switch from storing transposed SBP operators to regular operators) and then incorporate that change.

Reply:
Hm...it's a little more annoying than I thought to do this, since I assume skew-symmetry of the matrix. I've just precomputed `inv.(rd.wq)` so that this line is a multiply rather than a division.